### ISSUE TITLE
UX: Improve file actions menu

### ISSUE THREAD

**Issue by nokome** [stencila/hub#328]
The layout / organisation and labelling of the actions menu could be improved:
![image](https://user-images.githubusercontent.com/1152336/77274068-ec0b6080-6d19-11ea-99b9-8b054c9c942e.png)

Some initial thoughts:

- Icons next to each menu item could help make it a little less daunting
- Avoid the "Save as" repetition with a header (Bulma has menu headings I think)
- Drop the "as HTML" from Preview; put Preview next to Publish
- Download at the bottom?
- Do we need the latter part of "Open in Code Editor", "Open in Google Docs" etc.

@beneboy Could you do a very quick mock up of a new menu (even just a Markdown one with maybe a list of icon names) and post it back here for comments from me and @alex-ketch before going ahead and changing.

**Comment by beneboy** [comment]
Here's a few screenshots with the first icons that came to mind and quick rearrangements. Bulma has no section header so I just made it bold then indented the ones underneath.


![image](https://user-images.githubusercontent.com/292725/77302907-ef710d00-6d56-11ea-9e10-99d6c3d2a50f.png)
![image](https://user-images.githubusercontent.com/292725/77302944-fa2ba200-6d56-11ea-9697-9c8a24b55590.png)
![image](https://user-images.githubusercontent.com/292725/77302958-01eb4680-6d57-11ea-8f12-554cc1394b7b.png)

I think the *Save As* options should only be a couple of common ones and then *Other…* that opens a modal. JATS and JSON-LD don't strike me as being super popular. Another option might be a seperate Save As menu.

**Comment by nokome** [comment]
LGTM. 

Could you go ahead and implement the "Other… that opens a modal." bit but make if configurable so we can easily change what goes where by file type. e.g.

```js
SAVE_AS_FORMATS = {
  'gdoc': {
     'main': ['docx', 'md', 'ipynb'],
     'other': ['html', 'jats']
      ...
  },
  'csv': ...
}

FORMAT_LABELS = {
   'ipynb': 'Jupyter Notebook',
    'docx': 'Reproducible Word',
    'pdf': ' Reproducible PDF'
}

```

Unless that is going to take a long time, in which case maybe we will stick with the current version.

**Comment by beneboy** [comment]
I'm proposing for simplicity having a single modal that appears if you pick `Save as <type>` or `Save as other…`. We currently have a modal to enter the file name. I think we should just add a type dropdown to it, see WIP screenshot.

<img width="662" alt="image" src="https://user-images.githubusercontent.com/292725/77689013-dc379900-7005-11ea-843c-092f5d939d23.png">

If you select `Save as <type>` then the type is automatically selected, otherwise just the first type is selected. You could, of course `Save as HTML`  and then change the type in the modal to choose something else.

This would mean we wouldn't have to have two similar modals.

**Comment by nokome** [comment]
Yep, that makes sense +1

**Comment by nokome** [comment]
Closed in #348
